agent/xdsv2/rbac_resources.go

```diff
@@ -132,27 +133,25 @@ func makeRBACHTTPFilter(rbac *envoy_rbac_v3.RBAC) (*envoy_http_v3.HttpFilter, er
 	return makeEnvoyHTTPFilter(envoyHTTPRBACFilterKey, cfg)
 }
 
-func makeRBACPolicies(l4Permissions []*pbproxystate.Permission) map[string]*envoy_rbac_v3.Policy {
-	policyLabel := func(i int) string {
-		if len(l4Permissions) == 1 {
-			return baseL4PermissionKey
-		}
-		return fmt.Sprintf("%s-%d", baseL4PermissionKey, i)
-	}
-
+func makeL4RBACPolicies(l4Permissions []*pbproxystate.Permission) map[string]*envoy_rbac_v3.Policy {
 	policies := make(map[string]*envoy_rbac_v3.Policy, len(l4Permissions))
 
 	for i, permission := range l4Permissions {
-		policy := makeRBACPolicy(permission)
+		if len(permission.DestinationRules) != 0 {
+			// This is an L7-only permission
+			// ports are split out for separate configuration before this point and L7 filters are configured separately
+			continue
+		}
+		policy := makeL4RBACPolicy(permission)
```

Comment:
nit: is there a possibility that permission is `nil` here? doesn't look like `makeL4RBACPolicy` handles that possibility either. This could `panic` when we are trying to do permision.Principals

Reply:
I don't think it can be in this loop over the range, but I will add a nil checks in the make*Policy functions in so they are better protected for re-use